core/src/test/scala/org/broadinstitute/dsde/rawls/billing/BpmBillingProjectLifecycleSpec.scala

```diff
@@ -568,63 +603,63 @@ class BpmBillingProjectLifecycleSpec extends AnyFlatSpec {
+      new BpmBillingProjectLifecycle(mock[SamDAO],
+                                     repo,
+                                     bpm,
+                                     workspaceManagerDAO,
+                                     mock[WorkspaceManagerResourceMonitorRecordDao]
+      )
 
-    val result = bp.preDeletionSteps(
-      billingProjectName,
-      testContext
-    )
-    ScalaFutures.whenReady(result.failed) { exception =>
-      exception shouldBe a[BillingProjectDeletionException]
-      assert(exception.getMessage.contains(landingZoneErrorMessage))
+    val exception = intercept[BillingProjectDeletionException] {
+      Await.result(bp.initiateDelete(billingProjectName, testContext), Duration.Inf)
     }
+
+    assert(exception.getMessage.contains(landingZoneErrorMessage))
   }
 
   it should "succeed if the landing zone and billing profiles id do not exist" in {
     val repo = mock[BillingRepository]
     when(repo.getCreationStatus(billingProjectName)).thenReturn(Future.successful(CreationStatuses.Ready))
     when(repo.getLandingZoneId(billingProjectName)).thenReturn(Future.successful(None))
-    when(repo.getBillingProfileId(billingProjectName)).thenReturn(Future.successful(None))
 
     val bpm = mock[BillingProfileManagerDAO]
     val workspaceManagerDAO = mock[HttpWorkspaceManagerDAO]
     val bp =
-      new BpmBillingProjectLifecycle(repo, bpm, workspaceManagerDAO, mock[WorkspaceManagerResourceMonitorRecordDao])
+      new BpmBillingProjectLifecycle(mock[SamDAO],
+                                     repo,
+                                     bpm,
+                                     workspaceManagerDAO,
+                                     mock[WorkspaceManagerResourceMonitorRecordDao]
+      )
 
-    Await.result(bp.preDeletionSteps(
-                   billingProjectName,
-                   testContext
-                 ),
-                 Duration.Inf
-    )
+    Await.result(bp.initiateDelete(billingProjectName, testContext), Duration.Inf)
 
-    verify(workspaceManagerDAO, Mockito.times(0))
-      .deleteLandingZone(ArgumentMatchers.any[UUID], ArgumentMatchers.eq(testContext))
-    verify(bpm, Mockito.times(0)).deleteBillingProfile(ArgumentMatchers.any[UUID], ArgumentMatchers.eq(testContext))
+    verify(workspaceManagerDAO, Mockito.never()).deleteLandingZone(ArgumentMatchers.any(), ArgumentMatchers.any())
+    verify(bpm, Mockito.never()).deleteBillingProfile(ArgumentMatchers.any[UUID], ArgumentMatchers.eq(testContext))
   }
 
   it should "delete the landing zone if the id exists" in {
     val repo = mock[BillingRepository]
     when(repo.getCreationStatus(billingProjectName)).thenReturn(Future.successful(CreationStatuses.Ready))
     when(repo.getLandingZoneId(billingProjectName)).thenReturn(Future.successful(Some(landingZoneId.toString)))
-    // Mock no associated billing profile to delete
-    when(repo.getBillingProfileId(billingProjectName)).thenReturn(Future.successful(None))
```

Comment:
Why is this not needed?

Reply:
This was originally a test in the orchestrator. By moving it to the lifecycle, it narrows the scope of what we're testing, and eliminates the need to mock that call.